pkg/phlaredb/profiles.go

```diff
@@ -136,6 +136,15 @@ type profileSeries struct {
 	profilesOnDisk []*rowRange
 }
 
+type PersistentProfileSeriesCallback interface {
+	// PreCreationSeries is called before creating a series to indicate if the series can be created.
+	// A non nil error means the series should not be created.
+	PreCreationSeries(labels.Labels) error
+	// PreCreationProfiles is called before creating profiles to indicate if the profile can be created.
+	// A non nil error means the profile should not be created.
+	PreCreationProfiles(ps *schemav1.Profile, lbs phlaremodel.Labels) error
+}
+
```

Comment:
Not too sure how this is to be used? Naming is not great super clear, maybe it PreProfileCreationCheck() and PreSeriesCreationCheck() is better, probably shouldn't be there until we use it

Reply:
Yeah this is a left over. I'll remove it. Thanks.